config/configgates/README.md

```diff
@@ -0,0 +1,92 @@
+# Collector Feature Gates
+
+This package provides a mechanism that allows operators to enable and disable
+experimental or transitional features at deployment time. These flags should
+be able to govern the behavior of the application starting as early as possible
+and should be available to every component such that decisions may be made
+based on flags at the component level.
+
+## Usage
+
+A component that needs a feature gate that will not be shared with other 
+components should define a `Gate` as an unexported package variable and add 
+it to the global registry in an `init()` function.
+
+```go
+var fancyNewFeatureGate = &configgate.Gate{
+	ID:          "namespaced.uniqueIdentifier",
+	Description: "A brief description of what the gate controls",
+	Enabled:     false,
+}
+
+func init() {
+	err := configgates.GetRegistry().Add(fancyNewFeatureGate)
+	if err != nil {
+		panic(err)
+	}
+}
+```
+
+The `Enabled` property of the feature gate may then later be checked to 
+determine whether to enable the gated feature:
+
+```go
+if fancyNewFeatureGate.Enabled {
+	setupNewFeature()
+	doThingWithNewFeature()
+} else {
+	doThingWithOldFeature()
+}
+```
+
+For components that need to query the state of globally-applicable feature 
+gates a read-only implementation of the `configgates.Gates` interface is 
+available in the `*CreateSettings` structs passed to component factories. It 
+exposes an `IsEnabled(id string) bool` method that can be used to query the 
+status of any `Gate`.
+
+```go
+if settings.Features.IsEnabled("global.featureGate") {
```

Comment:
I wonder if we need this additional way of querying the gate instead of allowing to simply declare the gate anywhere as an exported variable (e.g. in a separate package) and query it directly? What do we gain by this?

Reply:
This becomes the only way of querying the state of a configured gate.  We gain the ability to separate configuration for `Service` instances and allow them to operate in the same process with independent gate configurations.